Public/Src/Cache/ContentStore/DistributedTest/Stores/GrpcCopyContentTests.cs

```diff
@@ -32,12 +32,67 @@ public class GrpcCopyContentTests : TestBase
     {
         private const int FileSize = 1000;
         private const HashType DefaultHashType = HashType.Vso0;
+        private const string LocalHost = "localhost";
         private Context _context;
+        private GrpcCopyClientCache _clientCache;
 
         public GrpcCopyContentTests()
             : base(() => new PassThroughFileSystem(TestGlobal.Logger), TestGlobal.Logger)
         {
             _context = new Context(Logger);
+            _clientCache = new GrpcCopyClientCache(_context);
+        }
+
+        [Fact]
+        public void DuplicateClientsAreTheSameObject()
+        {
+            using (var client1 = _clientCache.Create(LocalHost, 10, true))
+            using (var client2 = _clientCache.Create(LocalHost, 10, true))
+            {
+                Assert.Same(client1, client2);
+            }
+        }
+
+        [Fact]
+        public void ValidateBackgroundCleanup()
+        {
+            var key = new GrpcCopyClientKey(LocalHost, 11, true);
+            using (var client = _clientCache.Create(key.Host, key.GrpcPort, key.UseCompression))
+            { 
+                client._lastUseTime = DateTime.UtcNow - TimeSpan.FromHours(2);
+            }
+
+            _clientCache.StartBackgroundCleanup();

```

Comment:
I'm not sure we need it, right? We start it during startup anyways.

Reply:
The intent here is to force cleanup to happen now. We don't want to wait. I'll add a comment.

---
In reply to: [290518317](https://github.com/microsoft/BuildXL/pull/391#discussion_r290518317) [](ancestors = 290518317)